src/lib/api.js

```diff
@@ -1,6 +1,47 @@
 import axios from 'axios';
 
 const api = {
+    responseInterceptorId: null,
+    requestInterceptorId: null,
+    /**
+     * Adds a function to Axios that intercepts a response
+
+     * @public
+     * @param {Function} responseInterceptor - Function that gets called on each response
+     * @return {void}
+     */
+    addResponseInterceptor(responseInterceptor) {
+        if (responseInterceptor && typeof responseInterceptor === 'function') {
```

Comment:
should we add logic to eject any preexisting interceptors in case `this.*InterceptorId` is not null?

Reply:
It looks like `.use` pushes the interceptor onto the array rather than replacing it. It looks like there's a `.forEach` that we may be able to use to avoid tracking these IDs, though.

Reply:
Yeah, we can just iterate through the `Interceptor manager` and flush all of them. I'll implement that tomorrow.